### ISSUE TITLE
How to plot uncertainties on maps

### ISSUE THREAD

**Issue by kvelleby** [prio-data/prediction_competition_2023#29]
One option is to use alpha levels to plot uncertainty on a map, as in here: https://nhess.copernicus.org/articles/22/1499/2022/#&gid=1&pid=1 especially Figure 3. I am exploring this with Xiaolong.

_Originally posted by @paolavesco in https://github.com/prio-data/prediction_competition_2023/issues/22#issuecomment-1761065655_

**Comment by kvelleby** [comment]
It is also possible to [add hatching](https://stackoverflow.com/questions/71435447/how-to-add-hatching-by-condition-with-geopandas-and-matplotlib) to maps.

**Comment by kvelleby** [comment]
I am not sure what you mean about uncertainty, particularly in the context of mapping the error metrics.

We can plot the confidence of predictions (e.g., the highest_density/credibility/confidence interval of the prediction samples at any given observation).
We can explore the sensitivity of error metrics (but this is currently not done). See #19 for a discussion.
We can plot the mean ignorance scores converted to a probability (np.exp(-IGN)). This would tell us how high probability the model assigned to the actual outcome on average.

**Comment by paolavesco** [comment]
Yes I was thinking of the confidence of predictions, not the error metrics here.

**Comment by kvelleby** [comment]
Ok. One approach could then be to use `plotting.get_quantiles(low=0.25, high=0.75)` and plot `df.high - df.low` using `plotting.choropleth().` (The low and high percentiles could be other numbers of course.) This could be plotted as a separate plot. Sara can be working on this, I think.

If you rather would want the best estimate plotted, but hatching or something else whenever the confidence spans a wide range, we would need to define what kind of range would constitute an "uncertain" range. My initial thought is that this range should be defined globally, so that it is comparable across models. I think some testing would be needed to find a suitable range. Ideas and comments are welcome.

**Comment by paolavesco** [comment]
I think the first approach you suggest works well, with perhaps  0.95 or 0.99 too (in some cases, the interesting trends are shown there I think). Sara can coordinate with Xiaolong then, he had already started exploring options on this last week.

**Comment by xiaolong0728** [comment]
I am not so sure about the first approach. In the paper, they decided the uncertainty level using a random number generator (weird to me).  But in our case, does it mean we determine the uncertainty by the interval of `df.high - df.low`, i.e., the larger the interval, the larger the uncertainty? 
Also, we should combine the uncertainty with the actual fatalities map, right?
